src/machine/serial.go

```diff
@@ -44,3 +47,14 @@ func (ns NullSerial) Buffered() int {
 func (ns NullSerial) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
+
+// Read is a no-op; always returns that zero bytes were read with no error
+func (ns NullSerial) Read(p []byte) (n int, err error) {
+	return 0, nil
```

Comment:
Just wondering, shouldn't it return io.EOF to show that there is nothing to read?

As per doc
> // Implementations of Read are discouraged from returning a
// zero byte count with a nil error, except when len(p) == 0.
// Callers should treat a return of 0 and nil as indicating that
// nothing happened; in particular it does not indicate EOF.

The same as `ReadByte()` always returns an error

https://github.com/tinygo-org/tinygo/blob/3458726234bdd758cca3d55cc72af83954f6379d/src/machine/serial.go#L32-L34

Reply:
Agreed, returning an error seems more correct to me.

Reply:
Do you think should return `io.EOF` or should it return `errNoByte` (the same as `ReadByte`)?